rfcs/icon-component.md

```diff
@@ -0,0 +1,69 @@
+# Icon Component
+
+This is the set of ~16 icons used internally within baseui that are safe to open source. This is not the set of uber-specific icons published internally as @uber/icons (though the API is intended to align with that to reduce confusion).
+
+![image](https://user-images.githubusercontent.com/875591/45131331-51e8c500-b141-11e8-898d-ead83d44b345.png)
+
+## Usage
+
+```javascript
+import * as React from 'react';
+import {LeftArrow} from 'baseui/icon/left-arrow';
+
+export default () => (
+  <LeftArrow size="10px" color="#ccc" />
+);
+```
+
+## Exports
+
+* `Icon` (Base Icon component used by other exports below)
+
+From left-to-right based on the image above:
+
+* `DeleteAlt`
+* `Search`
+* `Delete`
+* `Check`
+* `ArrowUp`
+* `ArrowDown`
+* `ArrowLeft`
+* `ArrowRight`
+* `TriangleRight`
+* `TriangleDown`
+* `Upload`
+* `Plus`
+* `Grab`
+* `Alert`
+* `Menu`
+* `ChevronRight`
+* `ChevronLeft`
+* `Spinner`
+* `Overflow`
+* `TriangleLeft`
+* `TriangleUp`
+* `Filter`
+* `CheckIndeterminate`
+
+## `Icon` API
+
+* `size: number|string` - Optional, Defaults to `scale500` (12px)
+  Size of element, will be passed to the svg width/height style. Can also be a baseui `theme.sizing` constant like `sizing500`.
+* `color: string` - Optional, Defaults to `currentColor`
+  Color of icon, will be used as svg fill
+* `title: string` - Optional
+  Allows you to override the SVG `<title>` label, which is used for accessibility
```

Comment:
Should we just default to `Plus` and `ArrowLeft` etc? I think developer might forget to change this.

Reply:
yeah, we'll provide a default name for each component. this prop is mainly available for override in case the name title doesn't accurately fit the specific use case for the icon.